### PR TITLE
docs: new cache example and file name fixes

### PR DESCRIPTION
**Pull request by professor91**
- Added example for using cache
- Replaced - with _ in filenames and updated in respective md files
- Moved permissions example md file of Misc category to /misc dir

- [x] My pull request is made against the `dev` branch.
- [x] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] I tested my commits, by adding a test case to the unit tests if needed
- [x] I have ensured that I did not break any existing API calls.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.

### CODE REVIEW

**Review by braindigitalis — CHANGES_REQUESTED** [review]
not sure there's any need to rename all the images apart from that looks good 👍

**Comment by professor91** [comment]
updated

File: docpages/example_programs/the_basics/using_cache.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Below is an example showing how to get a user from the cache
+
+\include{cpp} using_cache.cpp 
+
+DPP caches more than just users, which you can get using the below-mentioned functions:
+- `dpp::find_role()`
+- `dpp::find_channel()`
+- `dpp::find_emoji()`
+- `dpp::find_guild()`
+- `dpp::find_guild_member()`
```
Comment (Jaskowicz1) [inline docpages/example_programs/the_basics/using_cache.md:18]: Needs a new line after this!

**Review by braindigitalis — CHANGES_REQUESTED** [review]
can you please also revert the name changes of the markdown files?
they don't need to be changed

**Comment by professor91** [comment]
> can you please also revert the name changes of the markdown files? they don't need to be changed

i changed so they'd in consistent format...the links don't change if that's your concern, lmk if they still need to be revert back

**Comment by braindigitalis** [comment]
> > can you please also revert the name changes of the markdown files? they don't need to be changed
> 
> i changed so they'd in consistent format...the links don't change if that's your concern, lmk if they still need to be revert back

yeah please revert those as it makes the PR shorter

**Comment by professor91** [comment]
added new line and reverted markdown filenames

File: docpages/example_programs/the_basics/using_cache.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+\page using-cache Using Cache
+
+Sometimes you may need information that is not directly available in the event callback object.
+
+To handle this DPP maintains a cache of commonly used data for you.
+
+@note As of August 30th, 2022, Discord made Guild Members a privileged intent. You must have GUILD_MEMBERS intent enabled for your app from discord developers portal to be able to look for members in cache.
```
Comment (braindigitalis) [inline docpages/example_programs/the_basics/using_cache.md:7]: this isn't technically true, id you don't have the intent the cache will "trickle fill" slowly with members as the bot sees them, e.g. when they send a message, join a VC, or send a slash command. you might also want to mention how slash commands have resolved sets that eliminates the need to look in the cache for data.

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**